### ISSUE TITLE
XML declaration lost

### ISSUE THREAD

**Issue by jirubio** [thymeleaf/thymeleaf#150]
I have a xml template with this as the first line:

```
<?xml version="1.0" encoding="UTF-8"?>
```

When using thymeleaf in Xml mode the encoding attribute disappears in the output.
After rendering the first line of the output is:

```
<?xml version="1.0"?>
```

This encoding information is important because the reader must know what was the encoding used for generating the text. This is mitigated in a web environment, the encoding can be retrieved from the content-type header, but in an offline environment there is no clue about the encoding.
Same happens for other attributes like "standalone".

**Comment by danielfernandez** [comment]
Yes, unfortunately the standard parsers _eat_ the XML declaration in XML files, so there actually is no _easy_ (or, more accurately, _non-extremely-difficult_) way for thymeleaf to recover the original XML decl.

This will be solved in version 3.0, when the current parsing system is subtituted by attoparser [ http://www.attoparser.org ]. But unfortunately, that won't be soon :-( (thymeleaf 2.1 will go first).

**Comment by jirubio** [comment]
As a workaround the first lines of the document could be parsed "manually", without depending on any external dependencies, at least until the version 3.0 arrives.

**Comment by danielfernandez** [comment]
Already included in `3.0.0-SNAPSHOT`
